lib/ansible/modules/cloud/amazon/iam_managed_policy.py

```diff
@@ -163,76 +183,99 @@ def get_or_create_policy_version(iam, policy, policy_document):
     # Instead of testing the versions list for the magic number 5, we are
     # going to attempt to create a version and catch the error
     try:
-        return iam.create_policy_version(
-            PolicyArn=policy['Arn'],
-            PolicyDocument=policy_document
-        )['PolicyVersion'], True
-    except Exception as e:
-        delete_oldest_non_default_version(iam, policy)
-        return iam.create_policy_version(
-            PolicyArn=policy['Arn'],
-            PolicyDocument=policy_document
-        )['PolicyVersion'], True
-
-
-def set_if_default(iam, policy, policy_version, is_default):
+        version = iam.create_policy_version(PolicyArn=policy['Arn'], PolicyDocument=policy_document)['PolicyVersion']
+        return version, True
+    except botocore.exceptions.ClientError as e:
+        delete_oldest_non_default_version(module, iam, policy)
```

Comment:
@dkhenry I've left the logic here pretty much as it was, but I don't really understand how this section of code is designed to work. Is there a limit to the number of versions allowed (is that what the 5 in your comment refers to?)